### DISCUSSION TITLE
Move otelkit to go-kit

### DISCUSSION THREAD

**Discussion by pellared** [go-kit/kit#1266] [General]
Hello 👋 

I am one of OpenTelemetry Go maintainers. 

Recently I was reviewing https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4044 and I started to think that it may be more beneficial for the community if the go-kit instrumentation library would be moved to this repository.

The reasons are the following

- the instrumentation is VERY go-kit specific it does not follow any [OpenTelemetry Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/) (e.g. for HTTP)
- OTel maintainers and approvers are on low capacity and lack expertise on go-kit. See: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/9437f9bf6a302c0d4728e9047894ba7c1de3bb72/CODEOWNERS#L36
- other tracing instrumentation can be found under https://github.com/go-kit/kit/tree/master/tracing
- this is currently a more preferred way according to our [recommendation](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/README.md#new-instrumentation)

Do you have any opinions?

CC @sagikazarmark

**Comment by peterbourgon** [comment]
It's unlikely that Go kit would agree to maintain new functionality like the package proposed here. But if you don't have the bandwidth for it in OTel, please don't feel obliged to keep it around. Mark it as deprecated and be free, no hard feelings 😉

**Comment by sagikazarmark** [comment]
@pellared thanks for reaching out.

Back when I added the instrumentation to that repo it was the recommendation. I guess due to otel still being young.

I have no problem moving that package out of the common repo though.

To be completely honest I will have to check if it's still useful before deciding if I want to host it anywhere at all.

@peterbourgon what do you think about moving it to a new repo?
